src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -218,7 +219,28 @@ internal async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsync(bool
                             targetsList = await GetTargetsListAsync(assetsFilePath, token);
                         }
 
-                        transitiveOrigins = calculatedTransitivePackages.Any() ? ComputeTransitivePackageOrigins(calculatedInstalledPackages, targetsList, token) : new Dictionary<string, TransitiveEntry>();
+                        // If the project has project references, we need to compute transitive origins for their packages
+                        List<LockFileTargetLibrary> projectReferences = targetsList.SelectMany(t => t.Libraries).Where(l => l.Type == LibraryType.Project).ToList();
+                        List<PackageReference> calculatedLibraryReferences = new List<PackageReference>(projectReferences.Count + calculatedInstalledPackages.Count);
+                        if (projectReferences.Any())
```

Comment:
good practice to compare count since you have a list.

Reply:
honestly, we should probably add `.Any()` (the overload without a predicate) to the banned API list.

Reply:
> honestly, we should probably add `.Any()` (the overload without a predicate) to the banned API list.
> 
> Although the way that `calculatedTransitivePackages` is used elsewhere, `Any()` is probably less costly in that specific use case than materializing a collection just so `.Count > 0` can be used.

I don't mind that.